src/lib/comms/sol-coap.c

```diff
@@ -1746,40 +1747,30 @@ network_event(void *data, const struct sol_network_link *link, enum sol_network_
 }
 
 static struct sol_coap_server *
-sol_coap_server_new_full(enum sol_socket_type type, const struct sol_network_link_addr *servaddr)
+sol_coap_server_new_full(struct sol_socket_ip_options *options, const struct sol_network_link_addr *servaddr)
```

Comment:
yes, it will come in the next pr. I'm doing it together with the oic over ble in order to have on more client of this api and see the best way to change it.